src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngle.java

```diff
@@ -1,35 +1,47 @@
 package edu.greenblitz.robotName.commands.swerve;
 
-import edu.greenblitz.robotName.subsystems.swerve.Chassis.ChassisConstants;
+import edu.greenblitz.robotName.FieldConstants;
 import edu.greenblitz.robotName.subsystems.swerve.Chassis.SwerveChassis;
+import edu.wpi.first.math.geometry.Pose2d;
 import edu.wpi.first.math.geometry.Rotation2d;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.function.Supplier;
+
+public class RotateToAngle extends SwerveCommand {
+
+    private Supplier<Rotation2d> angleSetPointSupplier;
 
-public class RotateToAngle extends SwerveCommand{
     private Rotation2d angleSetPoint;
 
-    public RotateToAngle(Rotation2d angleSetPoint){
+    public RotateToAngle(Rotation2d angleSetPoint) {
         super();
-        this.angleSetPoint = angleSetPoint;
+        this.angleSetPointSupplier = () -> angleSetPoint;
+    }
+
+    public RotateToAngle(Supplier<Rotation2d> angleSetPointSupplier) {
+        super();
+        this.angleSetPointSupplier = angleSetPointSupplier;
     }
 
     @Override
-    public void initialize(){
-        ChassisConstants.ROTATION_PID_CONTROLLER.setSetpoint(angleSetPoint.getRadians());
+    public void initialize() {
+        angleSetPoint = angleSetPointSupplier.get();
+        SwerveChassis.getInstance().setGoalAngle(angleSetPoint);
```

Comment:
if its a swerve command use the swerve object that is saved in the SwerveCommand